src/inputfetcher.h

```diff
@@ -0,0 +1,131 @@
+// Copyright (c) 2025-present The Bitcoin Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#ifndef BITCOIN_INPUTFETCHER_H
+#define BITCOIN_INPUTFETCHER_H
+
+#include <coins.h>
+#include <logging.h>
+#include <primitives/transaction_identifier.h>
+#include <txdb.h>
+#include <util/hasher.h>
+
+#include <cstdint>
+#include <stdexcept>
+#include <unordered_set>
+#include <utility>
+#include <vector>
+
+/**
+ * Helper for fetching inputs from the CoinsDB and inserting into the CoinsTip.
+ *
+ * It loops through the block and writes all input indexes to a
+ * vector. It then assigns itself an input from the vector, and
+ * fetches the coin from disk. The outpoint and coin pairs are written to a
+ * a different vector. Once all inputs are fetched, it writes the coins to the cache.
+ */
+class InputFetcher
+{
+private:
+    /**
+     * The flattened indexes to each input in the block. The first item in the
+     * pair is the index of the tx, and the second is the index of the vin.
+     */
+    std::vector<std::pair<size_t, size_t>> m_inputs{};
```

Comment:
the `vin` index can likely be a `uint32_t` instead
https://github.com/bitcoin/bitcoin/blob/e744fd1249bf9577274614eaf3997bf4bbb612ff/src/primitives/transaction.h#L32

Reply:
Done, using uint32_t in the Input struct now, not sure if it will have any effect in the struct though.